<?xml version="1.0"?>
<launch>
    
    <!--Arguments-->
    <arg name="world_name" default="threeLanes" />
    <arg name="gui" default="true" />

    <arg name="r_name" default="racecar"/>
    <arg name="vesc_ns" default="/vesc"/>
    <arg name="vis_laser" default="true"/>

    <arg name="r_type" default="racecar"/>
    <arg name="x_pos" default="0"/>
    <arg name="y_pos" default="-0.2625"/>
    <arg name="z_pos" default="0.05"/>
    <arg name="ctrl_method" default=""/>


    <!--Launching Gazebo With The Racecar-->
    <include file="$(find racecar_gazebo)/launch/one_racecar.launch">
      <arg name="world_name" value="$(arg world_name)" />
      <arg name="gui" value="$(arg gui)" />

      <arg name="r_name" value="$(arg r_name)"/>
      <arg name="vesc_ns" value="$(arg vesc_ns)"/>
      <arg name="vis_laser" value="$(arg vis_laser)"/>

      <arg name="r_type" value="$(arg r_type)"/>
      <arg name="x_pos" value="$(arg x_pos)"/>
      <arg name="y_pos" value="$(arg y_pos)"/>
      <arg name="z_pos" value="$(arg z_pos)"/>
      <arg name="ctrl_method" value="$(arg ctrl_method)"/>
    </include>


    <!-- Front steering: locate base_link at the center of the rear-axle -->
    <!-- For the cost planner -->
    <!-- See this https://answers.ros.org/question/237035/steering-axis-of-carlike-robot-with-teb_local_planner/ -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_to_rear_axle"
        args="-0.155 0.0 0.0 0.0 0.0 0.0 1.0 $(arg r_name)/base_link $(arg r_name)/rear_axle" />


    <!--- Run map server and AMCL -->
    <include file="$(find racecar_localization)/launch/one_racecar_amcl.launch">
      <arg name="r_name" value="$(arg r_name)"/>
      <arg name="vesc_ns" value="$(arg vesc_ns)"/>  
      <arg name="map_file" value="$(find racecar_mapping)/maps/$(arg world_name).yaml"/>
    </include>


    <!--The Move Car Node-->
    <include file="$(find racecar_move_car)/launch/racecar_move_car.launch">
      <arg name="r_name" value="$(arg r_name)"/>
      <arg name="vesc_ns" value="$(arg vesc_ns)"/>
    </include>

    <!--Action generator Node-->
    <!--<include file="$(find racecar_qlearning)/launch/action_generator.launch">
    <arg name="r_name" value="$(arg r_name)"/>
    </include>-->

    <!--Rviz Node-->
    <node name="$(anon rviz)" pkg="rviz" type="rviz"
          args="-d $(find racecar_rviz)/rviz/one_navigation.rviz" output="screen" ns="$(arg r_name)"/>


</launch>
